packages/core/src/sync-realtime/filter.ts

```diff
@@ -189,6 +189,13 @@ export const isTraceFilterMatched = ({
     return false;
   }
 
+  if (
+    filter.includeReverted === false &&
+    (trace.revertReason !== undefined || trace.error !== undefined)
+  ) {
+    return false;
+  }
```

Comment:
It's not super clear, this function returns true for the case "should this trace be saved in the database" and not "does this trace exactly match the filter such that it will cause an indexing function to run". The reason for the distinction is that we don't want the cache to be completely useless when the filter type slightly changes. For that reason this check should be removed. 

In this example, if the user decides that they want to include reverted traces after they have already fully synced their app, every block will need to be resynced.

Maybe we could add some comments acknowledging the properties that are ignored.